pkg/clustermesh/internal/remote_cluster.go

```diff
@@ -251,6 +255,10 @@ func (rc *remoteCluster) getClusterConfig(ctx context.Context, backend kvstore.B
 		}
 	}
 
+	rc.mutex.Lock()
+	rc.config = &models.RemoteClusterConfig{Required: requireConfig}
+	rc.mutex.Unlock()
+
```

Comment:
I think these changes to the `restartRemoteCluster` and `getClusterConfig` should be explained in the commit message.

Reply:
I've extended the commit message with the additional information.